rust-runtime/aws-smithy-runtime/src/client/interceptors/http_checksum_required.rs

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_runtime_api::client::interceptors::error::BoxError;
+use aws_smithy_runtime_api::client::interceptors::{Interceptor, InterceptorContext};
+use aws_smithy_runtime_api::client::orchestrator::{HttpRequest, HttpResponse};
+use aws_smithy_runtime_api::config_bag::ConfigBag;
+use http::header::HeaderName;
+use md5::{Digest, Md5};
+
+/// Plugin that adds legacy MD5 checksums to requests. This will panic for streaming requests. Only
+/// non-streaming requests are supported.
+#[derive(Debug, Default)]
+pub struct HttpChecksumRequiredInterceptor {}
+
+impl HttpChecksumRequiredInterceptor {
+    /// Creates a new `HttpChecksumRequiredInterceptor`
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Interceptor<HttpRequest, HttpResponse> for HttpChecksumRequiredInterceptor {
+    fn modify_before_signing(
+        &self,
+        ctx: &mut InterceptorContext<HttpRequest, HttpResponse>,
+        _cfg: &mut ConfigBag,
+    ) -> Result<(), BoxError> {
+        let req = ctx.request_mut()?;
+        let data = req
+            .body()
+            .bytes()
+            .expect("checksum can only be computed for non-streaming operations");
+
+        let checksum = Md5::digest(data);
```

Comment:
Oh, interesting. This is the old checksum implementation before flexible checksums became a thing. That's why the S3 model doesn't have it anymore.

I think we still need to preserve it since it's part of the Smithy spec, but its test should move into the generic clients instead of S3.

Reply:
I still don't understand why the test isn't failing in man since S3 removed the attribute from their model.

Reply:
The flexible checksums implementation preserves the functionality, so the outdated test doesn't fail.